lib/internal/fs/promises.js

```diff
@@ -678,6 +695,17 @@ async function writeFile(path, data, options) {
   return PromisePrototypeFinally(writeFileHandle(fd, data), fd.close);
 }
 
+function isCustomIterable(obj) {
+  if (typeof obj !== 'object' || obj === null) {
+    return false;
+  }
+
+  return SymbolAsyncIterator in obj || (
+    SymbolIterator in obj &&
+    !ArrayIsArray(obj) &&
+    !isTypedArray(obj));
```

Comment:
@ocuto I think @ronag meant doing this, that seems reasonable to me – it adds two extra checks when `obj` is async iterable, but I doubt that would be noticeable in practice anyway.
```suggestion
  return isIterable(obj) && ! isArrayBufferView(obj) && typeof obj !== 'string';
```

Reply:
BTW, why do we exclude arrays from the current check? Shouldn't they be treated as regular iterable here? If there's a good reason to exclude them, we should at least document the exception.

Reply:
@aduh95 
Thanks for the review.

The reason is that it fails when you run the test (test-fs-append-file.js).
```
=== release test-fs-append-file ===
Path: parallel/test-fs-append-file
node:internal/process/promises:245
          triggerUncaughtException(err, true /* fromPromise */);
          ^
AssertionError [ERR_ASSERTION]: Missing expected exception.
    at /Users/hiroyuki/personal-workspace/node/test/parallel/test-fs-append-file.js:146:10 {
  generatedMessage: false,
  code: ‘ERR_ASSERTION’,
  actual: undefined,
  expected: { code: ‘ENOENT’, message: /no such file or directory/ },
  operator: ‘throws’
}
Command: out/Release/node /Users/hiroyuki/personal-workspace/node/test/parallel/test-fs-append-file.js
[00:00|% 100|+   0|-   1]: Done
```

In this case, should I modify the test?

```
[false, 5, {}, [], null, undefined].forEach(async (data) => {
```
to
```
[false, 5, {}, null, undefined].forEach(async (data) => {
```

Reply:
I'd say we should remove the failing test case: a missing exception is not a breaking change, and it'd be weird to allow any iterable except array.

Reply:
@aduh95 cc: @ronag 
okay, I fixed it.